### PR TITLE
Switch to the most maintained LSLForge fork

### PR DESCRIPTION
**Pull request by aiwaiwa**
```
python --version
Python 3.11.3
```

With this PR kwdb will be more up-to-date with the mostly maintained LSLForge fork (https://github.com/KoolLSL/lslforge).

After building against the most recent database, comparison with LSLForge shows the following differences, that could be potentially adopted with zero changes:

`Constants.hs`:
* Integer constants represented as binary flags (`0x01`) replacing pure numeric (`1`) and vice versa, kwdb being more consistent in which variant to use.
* `PI_BY_TWO` difference by `0.00000001`, kwdb being more accurate in rounding the number.
* `RAD_TO_DEG` difference by `0.0000005`, kwdb being slightly more precise.
* LSLForge missing a few of its own `llc` constants as replacement for numbers.
* Invalid UTF-8 endpoints for `JSON_*` represented as sequences of bytes vs one large hex number:
  * `Constant "JSON_ARRAY" (SVal "\xfdd2")` -> `Constant "JSON_ARRAY" (SVal "\xef\xb7\x92")`

`FuncSigs.hs`:
* kwdb names for function parameters being more accurate:
  *  ex. `index` vs `number` in `llDetected*` family of functions.
* kwdb missing comments for `llLinksetData*` functions. To match the quality of LSLForge in this regard, they would have to be added.
* Historically appended `\n` characters at the end of function comments are harmlessly stripped with this PR. Eclipse seems to not care.

It should be mentioned that LSLForge hasn't yet updated to the PBR constants and functions.

### CODE REVIEW

**Comment by Sei-Lisa** [comment]
This is a surprise. I never thought the LSLforge output would be getting any attention after the original author said they were not interested. Would you please add the missing descriptions so they can be on par?

Any negative effects with (SVal "\xfdd2") vs (SVal "\xef\xb7\x92")? They seem like different strings.

**Comment by aiwaiwa** [comment]
Thank you for the follow-up! I want to clarify that I am not associated with KoolLSL, and I'm uncertain about their interest in adopting kwdb. However, I've successfully compiled LSLForge with the kwdb output, incorporating the changes from the PR. I might reach out to them and gauge their thoughts on the matter.

Alternatively, I may consider forking the KoolLSL repository and maintaining it with the kwdb output, although I'm not an expert in Haskell.

I'll also include any missing descriptions. And remove the output, as I saw you've requested in the past.

Regarding the comparison of `(SVal "\xfdd2")` and `(SVal "\xef\xb7\x92")`, it appears to be acceptable, as the https://wiki.secondlife.com/wiki/JSON_ARRAY mentions URL encoding in this exact style.

**Comment by Sei-Lisa** [comment]
Thanks for the additions. I see inconsistencies with periods at the end of sentences. The description of `llClearLinkMedia` adds a dot; in others (e.g. `LINKSETDATA_RESET`) it's respected, and others (e.g. `llSetRenderMaterial`) remove it. I think it's best if the periods are present.

For all Python modules you're changing, please update the version. Please change the version of the database too - only the '<keywords version=...` line. The SL grid version should not change.

I've found that a previous PR introduced a duplicate constant, namely `LINKSETDATA_MULTIDELETE`. Would you be so kind to remove the duplicate as well?

**Comment by aiwaiwa** [comment]
I was about to ask if you'll be OK that I add the dots everywhere! I'll do what you've asked and ping you for a review.

**Comment by aiwaiwa** [comment]
Meant to say "periods" in the commit.

**Comment by Sei-Lisa** [comment]
I didn't expect you to go over all of them!

Technically there are grammar errors - periods theoretically should be placed before the parentheses. I'm generally OK with not respecting that rule because I feel it's illogical, but it's a problem in the case of `llKey2Name` where there's a dot before the parenthesis and another after it.

I think I'd prefer if you just remove that commit (`git rebase -i`) and merely add dots *only* on sentences you modified in the previous one. It's a bit unmanageable as-is and I'm probably missing things. I've also noticed some other minor problems and don't want this to be an eternal back-and-forth of reviews and commits.

Thank you for all your effort so far, by the way.

**Comment by aiwaiwa** [comment]
I've got an idea. I can fix grammar errors, there's no problem in back-and-forth for me, I am not rushing anywhere. If I just fix a few you noticed, you won't need to come back and add those periods again. It's a live organism anyway. What do you think?

**Comment by aiwaiwa** [comment]
I can randomly pick a style guide (APA, MLA, or Chicago), add a separate commit and you comment on those that feel wrong. If it goes wrong, I'll rebase.

**Comment by Sei-Lisa** [comment]
It's just more effort than I currently can afford to place into this. I really prefer that commit to be removed. With `git rebase -i master` you should be able to remove it. You may get a conflict with the commit that removes `LINKSETDATA_MULTIDELETE` but it should be easy to resolve. Follow carefully the instructions given by git, in case you're unfamiliar with these operations. You'll have to force-push it after it's fixed, with `git push -f origin update_lslforge`. Sorry if I sound patronizing, I don't know what you know.

The English grammar dictates that periods should be before the close parenthesis, not after. That's illogical in many cases in my opinion. I also found a couple OpenSim sentences that start in lower-case, can't remember which. I didn't finish reviewing, there's just too much to read.

On a different matter, I've looked into Haskell escaping rules, and it's not clear to me at all that either `\xfdd2` or `\xef\xb7\x92` is correct. I think that testing the optimization of a string containing one of these constants is a must.

And please don't forget to bump the version of the modified Python files.

**Comment by aiwaiwa** [comment]
Sure, no problem. I don't like long PRs either :D

File: database/kwdb.xml [hunk 1]
```diff
@@ -10161,7 +10161,7 @@ Returns the type constant for the value in json indicated by specifiers
   <function name="llReturnObjectsByID" type="integer" grid="sl">
     <param type="list" name="objects"/>
     <description lang="en">
-Returns an integer that is the number of objects successfully returned to their owners or an ERR_* flag.llReturnObjectsByOwner
+Returns an integer that is the number of objects successfully returned to their owners or an ERR_* flag.llReturnObjectsByOwner.
```
Comment (Sei-Lisa) [inline database/kwdb.xml]: I don't think `llReturnObjectsByOwner` should appear there in the first place.

Comment (Sei-Lisa) [reply database/kwdb.xml]: Also, the description talks about the return value, but not about what the function does. Something along these lines would work better, I think:

> Returns to their owners the objects with the ids given in a list. The return value is the number of objects successfully returned, or an ERR_* flag.

And something similar for `llReturnObjectsByOwner`.

Comment (aiwaiwa) [reply database/kwdb.xml]: Good catch! Fixing

**Comment by aiwaiwa** [comment]
> Any negative effects with (SVal "\xfdd2") vs (SVal "\xef\xb7\x92")? They seem like different strings.

Not a problem anymore, producing exactly the same thing now.

LSLForge is successfully compiling.

**Comment by Sei-Lisa** [comment]
Okay, it seems to be in good shape now. Thanks a lot for your collaboration!

**Pull request merged**